File: src/main/java/net/softbell/bsh/dto/view/general/MonitorSummaryCardDto.java
package net.softbell.bsh.dto.view.general;

import java.util.ArrayList;
import java.util.List;

import lombok.Getter;
import lombok.Setter;
import net.softbell.bsh.domain.entity.Node;

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 모니터뷰 카드정보 DTO
 */
@Getter
@Setter
public class MonitorSummaryCardDto
{
	private String alias;
	private List<MonitorCardItemDto> listItems;
	private Byte controlMode;
	// TODO 마지막 연결 시각
	
	public MonitorSummaryCardDto(Node entity)
	{
		// Exception
		if (entity == null)
			return;
		
		// Init
		listItems = new ArrayList<MonitorCardItemDto>();
		
		// Convert
		this.alias = entity.getAlias();
		this.controlMode = entity.getControlMode();
	}
}

File: src/main/kotlin/net/softbell/bsh/dto/view/general/MonitorSummaryCardDto.kt
package net.softbell.bsh.dto.view.general

import net.softbell.bsh.domain.entity.Node
import java.util.*

/**
 * @author : Bell(bell@softbell.net)
 * @description : 모니터뷰 카드정보 DTO
 */
class MonitorSummaryCardDto(entity: Node) {
    val alias: String = entity.alias
    val listItems: MutableList<MonitorCardItemDto> = ArrayList()
    var lastReceive: Date? = null
    var lastReceiveSecond: Long? = null
}